subsys/bootloader/bl_crypto_client/bl_crypto_client.c

```diff
@@ -21,20 +22,21 @@ const struct fw_info_abi *get_bl_crypto_abi(enum abi_index abi)
 		switch(abi) {
 		case BL_ROT_VERIFY:
 			bl_crypto_abis[abi] = fw_info_abi_find(
-					BL_ROT_VERIFY_ABI_ID,
+					CONFIG_BL_ROT_VERIFY_ABI_ID,
```

Comment:
Again, these switch cases looks like they could be preprocessed to have something like 

```
switch(abi): {
 CASE(BL_ROT_VERIFY);
 CASE(BL_SHA256);
 CASE(BL_SECP256R1;
}
```

Reply:
I'm not in favor of calling the macro CASE and have it contain `case` statements. Now you are changing the definition of the language even if it's upper case it'll decrease readability with few benefits.

Maybe:
```C
#define ABI_FIND(ABI_NAME) ABI_NAME: \
             bl_crypto_abis[abi] = fw_info_abi_find(CONFIG_ ## ABI_NAME ## _ABI_ID,\
                                                    CONFIG_ ## ABI_NAME ## _ABI_FLAGS,\
                                                    CONFIG_ ## ABI_NAME ## _ABI_VER,\
                                                    CONFIG_ ## ABI_NAME ## _ABI_MAX_VER);\
             break
```
Having it look like
```C
switch(abi): {
  case ABI_FIND(BL_ROT_VERIFY);
  case ABI_FIND(BL_SHA256);
  case ABI_FIND(BL_SECP256R1);
}
```
would be better.